backend/agent_logic/tools/sandbox_executor.py

```
@@ -1,79 +1,257 @@
-                "output": result.stdout.decode() if result.stdout else "",
-                "error": result.stderr.decode() if result.stderr else "",
-                "exit_code": result.returncode
+                "success": success,
+                "output": stdout,
+                "error": stderr,
+                "exit_code": exit_code,
+                "logs": logs
             }
             
+        except docker.errors.ContainerError as e:
+            logger.error(f"? Container execution error: {str(e)}")
```

Ruff (0.14.2)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)